p2p/src/peer.rs

```diff
@@ -175,8 +187,22 @@ mod run {
             let mut message_reader = MessageReader::new(read, cryptographer.clone());
             let mut message_sender = MessageSender::new(write, cryptographer);
 
+            let mut idle_interval = tokio::time::interval_at(Instant::now() + idle_timeout, idle_timeout);
+            let mut ping_interval = tokio::time::interval_at(Instant::now() + idle_timeout / 2, idle_timeout / 2);
+
             loop {
                 tokio::select! {
+                    _ = ping_interval.tick() => {
+                        iroha_logger::trace!("Sending ping");
```

Comment:
(probably needs a better way for displaying the duration)

```suggestion
                        iroha_logger::trace!("The connection has been idle for {:?}, pinging to check if it's alive", ping_interval.period());
```